llvm/lib/IR/DiagnosticInfo.cpp

```diff
@@ -159,7 +159,10 @@ std::string DiagnosticInfoWithLocationBase::getLocationStr() const {
   unsigned Column = 0;
   if (isLocationAvailable())
     getLocation(Filename, Line, Column);
-  return (Filename + ":" + Twine(Line) + ":" + Twine(Column)).str();
+  return (getAbsolutePath()
```

Comment:
don't use implicit `std::string` to bool conversion

Reply:
can we just always use `getAbsolutePath()`? or else we'll get inconsistent behavior

Reply:
Thanks for the comment, I've updated it accordingly which remove the std::string to bool conversion, and used getAbsolutePath() always when the file location can be found.